test/OpenTelemetry.Tests/Trace/TracerProvideSdkTest.cs

```diff
@@ -185,6 +186,95 @@ public void ProcessorDoesNotReceiveNotRecordDecisionSpan()
             Assert.False(endCalled);
         }
 
+        [Fact]
+        public void TracerProvideSdkCreatesActivitySource()
+        {
+            TestActivityProcessor testActivityProcessor = new TestActivityProcessor();
+
+            bool startCalled = false;
+            bool endCalled = false;
+
+            testActivityProcessor.StartAction =
+                (a) =>
+                {
+                    startCalled = true;
+                };
+
+            testActivityProcessor.EndAction =
+                (a) =>
+                {
+                    endCalled = true;
+                };
+
+            TestInstrumentation testInstrumentation = null;
+            using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                        .AddProcessor(testActivityProcessor)
+                        .AddInstrumentation((adapter) =>
+                        {
+                            testInstrumentation = new TestInstrumentation(adapter);
+                            return testInstrumentation;
+                        })
+                        .Build();
+
+            var adapter = testInstrumentation.Adapter;
+            Activity activity = new Activity("test");
+            activity.Start();
+            adapter.Start(activity);
+            adapter.Stop(activity);
+            activity.Stop();
+
+            Assert.True(startCalled);
+            Assert.True(endCalled);
+
+            TestActivityProcessor testActivityProcessorNew = new TestActivityProcessor();
+
+            bool startCalledNew = false;
+            bool endCalledNew = false;
+
+            testActivityProcessorNew.StartAction =
+                (a) =>
+                {
+                    startCalledNew = true;
+                };
+
+            testActivityProcessorNew.EndAction =
+                (a) =>
+                {
+                    endCalledNew = true;
+                };
+
+            tracerProvider.AddProcessor(testActivityProcessorNew);
```

Comment:
adding processor after build is supported, but due to bug 1075, instrumentation/activitysourceadapter doesn't get the newly added processor.